### PR TITLE
Set up module for Amplitude ingestion

### PR DESCRIPTION
**Pull request by angela-tran**
# Overall Description

This PR simply sets up the Python module that will contain our Airflow operator and the function for creating a DataFrame from Amplitude data. 

This is a base from which we can work on other issues for epic #960.

## Checklist for all PRs

- [x] Run `pre-commit run --all-files` to make sure markdown/lint passes
- [x] Link this pull request to all issues that it will close using keywords (see GitHub docs about [Linking a pull request to an issue using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). Also mention any issues that are partially addressed or are related.

### CODE REVIEW

**Comment by angela-tran** [comment]
@evansiroky Just wanted to tag you in case you have any thoughts or opinions on process as the Compiler team contributes code for #960. 😄 

My current thinking is that we'll open PRs for increments of work like this one and merge them into `main`. Our PRs should always have the codebase in a working state. If your team would rather have us follow a different strategy though, please do let us know!

**Review by thekaveman — DISMISSED** [review]

**Comment by evansiroky** [comment]
> @evansiroky Just wanted to tag you in case you have any thoughts or opinions on process as the Compiler team contributes code for #960. 😄
> 
> My current thinking is that we'll open PRs for increments of work like this one and merge them into `main`. Our PRs should always have the codebase in a working state. If your team would rather have us follow a different strategy though (like working off a feature branch or something), please do let us know!

I think it would be better to either do merges into a feature branch or just do all of the items at once like in #518. An added benefit of the approach with #518 was that it uses non-business data and is in the Sandbox DAG, so it serves as a good example for future efforts to read in actual business data as detailed in #774. It looks like the benefits team is able to provide a detailed review along the way if y'all choose to do the feature branch strategy. From my perspective I'm mainly concerned about the overall deliverable and whether the resulting DAG runs properly and won't crash the other DAGs.

File: airflow/plugins/operators/amplitude_to_warehouse.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"""Module for exporting data from Amplitude and adding to the data warehouse"""
+from airflow.models import BaseOperator
+
+
+def amplitude_to_df():
+    """Export data from Amplitude into a dataframe."""
+
```
[the remark this answers is not in the document]
Comment (angela-tran) [reply airflow/plugins/operators/amplitude_to_warehouse.py:7]: I thought that too. I learned that the one-line docstring counts as a statement, so `pylint` will flag `pass` here as an "Unnecessary pass statement".  
image

(When I had it as no docstring and just `pass`, then `pylint` flagged not having a docstring as a problem. 😐)

As a side note, I see that `pylint` is enabled in [devcontainer.json](https://github.com/cal-itp/data-infra/blob/9a3ebd69d767db46c666299e594d82c8aaade40c/.devcontainer/devcontainer.json#L23), but `.pre-commit-config.yaml` specifies [flake8](https://github.com/cal-itp/data-infra/blob/9a3ebd69d767db46c666299e594d82c8aaade40c/.pre-commit-config.yaml#L5) as the linter. Maybe I should change that line in devcontainer.json...? 🤷

[the remark this answers is not in the document]
Comment (thekaveman) [reply airflow/plugins/operators/amplitude_to_warehouse.py:7]: Weird that it didn't complain about line 20 - `pass` and no docstring?? 🤷 

Let's keep our changes hyper-focused on the Amplitude piece for now, but definitely open issues to get these QOL changes (e.g. with the linter difference) in the backlog.

[the remark this answers is not in the document]
Comment (angela-tran) [reply airflow/plugins/operators/amplitude_to_warehouse.py:7]: Looks like there is already an issue being worked on to make linting configuration more consistent: https://github.com/cal-itp/data-infra/issues/1025

**Comment by angela-tran** [comment]
@thekaveman @machikoyasuda I changed the target of this PR to a feature branch named `feat-amplitude-operator`, so this is now ready again for your review.

@evansiroky I unassigned you as reviewer - I am guessing you will be more interested in the final PR that will target `main`.

**Review by thekaveman — APPROVED** [review]

**Pull request merged**